### ISSUE TITLE
Could you release the training code and evaluation code about motion editing in the paper?

### ISSUE THREAD

**Issue by shuowang666** [exitudio/MMM#16]

**Comment by exitudio** [comment]
For temporal editing tasks, there is no need for training as it is an innate capability of bidirectional masked models. The sample code for editing tasks can be found in the edit.ipynb file of this repo.

It is evaluated by generating 50% of the sequences conditioned on the first and last 25% of the sequences. If you want the exact implementation of the evaluation function, you can email me directly.

**Comment by shuowang666** [comment]
I have emailed you with the subject of "request for evaluation code about motion editing in the paper". Could you provide the exact implementation of the evaluation function? Thank you very much for your help!

**Comment by exitudio** [comment]
Sorry for the late reply. We are upgrading the server. I will send the code to you when I can access the server.

**Comment by shuowang666** [comment]
Thank you for your reply!  Have you finished upgrading the server?

**Comment by exitudio** [comment]
I think I replied email to the wrong person. Maybe, I can put it here. I already upload to [gdrive](https://drive.google.com/file/d/1LZwdnnKzHYa-W6mKzwVuTfmYKyJnncSR/view?usp=drive_link)
Please see "edit_eval.py" 
"edit_task" has 4 types:
   edit_task = 'prefix' # inbetween, 'outpainting', prefix, suffix upperbody

**Comment by shuowang666** [comment]
Thanks very much for your help!

**Comment by shuowang666** [comment]
Thank you very much for the valuable code. 
It seems that the edit tasks provided in the code are in the condition with text, do I need to retrain the model for the validation of the edit task without any condition (e.g. text)?

**Comment by exitudio** [comment]
Yes. We retrained without text version.
